domaci_maze.py

```diff
@@ -90,495 +77,523 @@ def has_value(self) -> bool:
-        if row > 0 and self.board[row - 1, col].is_steppable():
-            potential_actions["up"] = (row - 1, col)
-        if row < self.board.rows_no - 1 and self.board[row + 1, col].is_steppable():
-            potential_actions["down"] = (row + 1, col)
-        if col > 0 and self.board[row, col - 1].is_steppable():
-            potential_actions["left"] = (row, col - 1)
-        if col < self.board.cols_no - 1 and self.board[row, col + 1].is_steppable():
-            potential_actions["right"] = (row, col + 1)
-
-        # # Randomly select a subset of these potential actions
-        # num_actions = randint(
-        #     1, len(potential_actions)
-        # )  # At least 1 action, up to the number of potential actions
-        # actions = dict(
-        #     sample(
-        #         list(potential_actions.items()),
-        #         min(num_actions, len(potential_actions)),
-        #     )
-        # )
-
-        return potential_actions
-
-    def validate_position(self, row, col):
-        """A utility function that validates a position."""
-        if row < 0 or row >= self.rows_no:
-            raise Exception("Invalid row position.")
-        if col < 0 or col >= self.cols_no:
-            raise Exception("Invalid column position.")
-        if not self.board[row, col].is_steppable():
-            raise Exception("Invalid position: unsteppable cell.")
-        return row, col
-
-    def move_from(self, row: int, col: int, action: str) -> tuple[int, int]:
-        if action in self.graph[self.board[row, col]]:
-            return self.graph[self.board[row, col]][action]
-        else:
-            raise Exception(f"Invalid action: {action} for cell ({row}, {col}).")
-
-    def __call__(
-        self, state: tuple[int, int], action: str
-    ) -> tuple[tuple[int, int], float, bool]:
-        row, col = state
-        new_row, new_col = self.move_from(row, col, action)
-        new_cell = self.board[new_row, new_col]
-        reward = new_cell.get_reward()
-        is_terminal = new_cell.is_terminal()
-        return (new_row, new_col), reward, is_terminal
-
-    def get_states(self):
-        states = []
-        for r in range(self.board.rows_no):
-            for c in range(self.board.cols_no):
-                if self.board[r, c].is_steppable():
-                    states.append((r, c))
-        return states
-
-    def is_terminal(self, s):
-        return self.board[s[0], s[1]].is_terminal()
-
-    def get_actions(self, cell):
-        if cell in self.graph:
-            return list(self.graph[cell].keys())
-        else:
+    def set_probabilities(self, node, g):
+
+        # terminal and wall node do not have any probabilities
+        if isinstance(node, WallNode) or isinstance(node, TerminalNode):
             return []
 
+        # list is easier to iterate, only because of that is node_list made
+        nodes_list = [node for node in g]

```

Comment:
Postoji to list nisi morao ovako sam manuelno. Ali ako radi kako treba top

Reply:
I ovo se slazem, ovo je samo peske